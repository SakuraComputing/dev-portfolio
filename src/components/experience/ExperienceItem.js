import React from 'react';

const ExperienceItem = (props) => {
    return (
        <div>
            <div className="item">
                <h3 className="title">{props.role.title} - <span className="place"><a
                    href="#">{props.role.company}</a></span> <span className="year">{props.role.year}</span></h3>
                <p>{props.role.description}</p>
            </div>
        </div>
    );
};

export default ExperienceItem;
