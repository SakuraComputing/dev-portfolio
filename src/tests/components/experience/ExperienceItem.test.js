import React from 'react';
import { shallow } from 'enzyme';
import ExperienceItem from '../../../components/experience/ExperienceItem';

describe('ExperienceItem Section', () => {
    it('should render correctly', () => {
        const experience = {
            id: 1,
            title: 'Dev',
            year: "(2016 - present)",
            company: "DHL",
            description: "Wibble"
        };
        const wrapper = shallow(<ExperienceItem experience={experience}/>);
        expect(wrapper).toMatchSnapshot();
    });
});