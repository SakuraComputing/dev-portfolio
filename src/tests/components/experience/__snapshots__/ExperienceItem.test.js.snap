// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceItem Section should render correctly 1`] = `
<div>
  <div
    className="item"
  >
    <h3
      className="title"
    >
      Co-Founder & Lead Developer - 
      <span
        className="place"
      >
        <a
          href="#"
        >
          Startup Hub
        </a>
      </span>
       
      <span
        className="year"
      >
        (2014 - Present)
      </span>
    </h3>
    <p>
      Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Donec vitae sapien ut libero venenatis faucibus. Nullam quis ante. Etiam sit amet orci eget eros faucibus tincidunt.
    </p>
  </div>
</div>
`;
